lib/ansible/plugins/connection/network_cli.py

```diff
@@ -342,15 +363,34 @@ def receive(self, command=None, prompts=None, answer=None, newline=True, prompt_
         '''
         Handles receiving of output from command
         '''
-        recv = BytesIO()
-        handled = False
-
         self._matched_prompt = None
         self._matched_cmd_prompt = None
+        recv = BytesIO()
+        handled = False
+        command_prompt_matched = False
         matched_prompt_window = window_count = 0
+        command_timeout = self.get_option('persistent_command_timeout')
 
         while True:
-            data = self._ssh_shell.recv(256)
+            if command_prompt_matched:
+                try:
+                    signal.signal(signal.SIGALRM, self._handle_buffer_read_timeout)
+                    signal.setitimer(signal.ITIMER_REAL, self.get_option('persistent_buffer_read_timeout'))
```

Comment:
This will raise `signal.ItimerError` if the timeout is < 0, and fails disastrously when timeout == 0

Reply:
Ok i can add a check to throw error is timeout value is <= 0